File: api/functional/flow-creator/src/main/java/com/emanuelvictor/api/functional/flowcreator/domain/entities/Option.java
package com.emanuelvictor.api.functional.flowcreator.domain.entities;

import com.emanuelvictor.api.functional.flowcreator.infrastructure.persistence.generic.PersistentEntity;
import lombok.Getter;
import lombok.Setter;


/**
 * @author Emanuel Victor
 * @version 1.0.0
 * @since 1.0.0, 25/08/2021
 */
@Setter
@Getter
@lombok.EqualsAndHashCode(callSuper = true)
public class Option extends PersistentEntity {

    /**
     *
     */
    private String name;

    /**
     * @param name String
     */
    public Option(final String name) {
        this.name = name;
    }
}

File: api/functional/flow-creator/src/main/kotlin/com/emanuelvictor/api/functional/flowcreator/domain/entities/option/Option.kt
package com.emanuelvictor.api.functional.flowcreator.domain.entities.option

import io.github.emanuelvictor.commons.persistence.generic.PersistentEntity


/**
 * @author Emanuel Victor
 * @version 1.0.0
 * @since 1.0.0, 25/08/2021
 */
abstract class Option(val identifier: String) : PersistentEntity()